commons-numbers-complex-arrays/src/main/java/org/apache/commons/numbers/complex/arrays/ComplexList.java

```diff
@@ -0,0 +1,357 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.commons.numbers.complex.arrays;
+
+import org.apache.commons.numbers.complex.Complex;
+
+import java.util.AbstractList;
+import java.util.Arrays;
+import java.util.Collection;
+
+/**
+ * Resizable-double array implementation of the List interface. Implements all optional list operations,
+ * and permits all elements. In addition to implementing the List interface,
+ * this class provides methods to manipulate the size of the array that is used internally to store the list.
+ *
+ * <p>Each ComplexList instance has a capacity. The capacity is half the size of the double array used to store the elements
+ * in the list. It is always at least twice as large as the list size. As elements are added to an ComplexList,
```

Comment:
The capacity is not `always at least twice as large as the list size` since capacity is referring to the number of elements, not doubles, that may be added before resizing.

This intro javadoc should describe the size limitation of the list. Since the data is stored using an interleaved format the maximum number of elements that may be added is approximately 2^30. This is half the maximum capacity of java.util.ArrayList. The memory usage is more efficient than using a List of Complex objects as the underlying numbers are not stored using instances of Complex.

The javadoc can be expanded to describe how to manipulate the numbers efficiently using the Complex functional interfaces when this functionality is added to the list.